src/internal_events/kafka.rs

```diff
@@ -0,0 +1,84 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct KafkaEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for KafkaEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "received one event.", rate_limit_secs = 10);
+    }
+
+    fn emit_metrics(&self) {
+        counter!(
+            "events_processed", 1,
+            "component_kind" => "source",
+            "component_type" => "kafka",
+        );
+        counter!(
+            "bytes_processed", self.byte_size as u64,
+            "component_kind" => "source",
+            "component_type" => "kafka",
+        );
+    }
+}
+
+#[derive(Debug)]
+pub struct KafkaOffsetUpdateFailed {
+    pub error: rdkafka::error::KafkaError,
+}
+
+impl InternalEvent for KafkaOffsetUpdateFailed {
+    fn emit_logs(&self) {
+        error!(message = "unable to update consumer offset.", error = %self.error);
+    }
+
+    fn emit_metrics(&self) {
+        counter!(
+            "consumer_offset_updates_failed", 1,
```

Comment:
This one is a bit arbitrary.

I'm not entirely sure when we want to add a counter for a certain event, and when we don't.

For example, I didn't add a `payload_extractions_failed` counter to the `KafkaPayloadExtractionFailed` event yet, so we'd either add that one as well, or remove this one to be consistent (or, we consider one more important than the other, in which case we should probably document this in the code).

cc @binarylogic @lukesteensen

Reply:
This seems reasonable to me. We don't have any solid rules around this kind of thing yet, but we can try to develop good patterns over time.